src/Sarif.Viewer.VisualStudio.Interop/SarifViewerInterop.cs

```diff
@@ -0,0 +1,190 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Sarif.Viewer.Interop
+{
+    public class SarifViewerInterop
+    {
+        private static readonly string ViewerAssemblyFileName = "Microsoft.Sarif.Viewer";
+        private static readonly string ViewerServiceInterfaceName = "SLoadSarifLogService";
+        private bool? _isViewerExtensionInstalled;
+        private bool? _isViewerExtensionLoaded;
+        private Assembly _viewerExtensionAssembly;
+        private AssemblyName _viewerExtensionAssemblyName;
+        private Version _viewerExtensionVersion;
+
+        /// <summary>
+        /// Gets the Visual Studio shell instance object.
+        /// </summary>
+        public IVsShell VsShell { get; private set; }
+
+        /// <summary>
+        /// Gets the unique identifier string of the SARIF Viewer package.
+        /// </summary>
+        public static readonly string ViewerExtensionGuidString = "b97edb99-282e-444c-8f53-7de237f2ec5e";
```

Comment:
>ViewerExtensionGuidString [](start = 38, length = 25)

Do we need this extra level of indirection? Is this used anywhere else? Would it suffice to inline it in the initialization of `ViewerExtensionGuid`?

Reply:
It's public so callers can use it. I guess the guid would provide this too.

---
In reply to: [211044522](https://github.com/Microsoft/sarif-visualstudio-extension/pull/9#discussion_r211044522) [](ancestors = 211044522)

Reply:
Yes, I'd expect consumers to consume the GUID, not the string.

---
In reply to: [211047444](https://github.com/Microsoft/sarif-visualstudio-extension/pull/9#discussion_r211047444) [](ancestors = 211047444,211044522)

Reply:
And my point wasn't just that it's public; I don't think it needs to exist at all. Just:
```
public static readonly Guid ViewerExtensionGuid = new Guid("b97e...");
```
would suffice.

---
In reply to: [211047778](https://github.com/Microsoft/sarif-visualstudio-extension/pull/9#discussion_r211047778) [](ancestors = 211047778,211047444,211044522)